tests/std/tests/GH_000690_overaligned_function/test.cpp

```diff
@@ -16,12 +16,12 @@ struct alignas(2 * alignof(std::max_align_t)) overaligned_t {
     char non_empty;
 
     void operator()(const void* const storage, const std::size_t storage_size) const {
-        const unsigned char* storage_bytes = reinterpret_cast<const unsigned char*>(storage);
-        const unsigned char* this_bytes    = reinterpret_cast<const unsigned char*>(this);
+        const std::uintptr_t storage_ptr_value = reinterpret_cast<std::uintptr_t>(storage);
+        const std::uintptr_t this_ptr_value    = reinterpret_cast<std::uintptr_t>(this);

```

Comment:
Pure style (so ignore if you like): I prefer to use `auto` when initializing a variable with the result of a cast rather than repeating the type name:
```suggestion
        const auto storage_ptr_value = reinterpret_cast<std::uintptr_t>(storage);
        const auto this_ptr_value    = reinterpret_cast<std::uintptr_t>(this);
```